editor/editor/layout.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Internal dependencies
+ */
+import ModeSwitcher from './mode-switcher';
+import EditorText from './mode/text';
+import EditorVisual from './mode/visual';
+
+class Layout extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.switchMode = this.switchMode.bind( this );
+		this.changeHtml = this.changeHtml.bind( this );
+		this.changeBlocks = this.changeBlocks.bind( this );
+		this.state = {
+			mode: 'visual',
+			html: '',
+			blocks: []
+		};
+	}
+
+	componentDidMount() {
+		this.setState( {
```

Comment:
At the very least we should try to do this in `componentWillMount` or the constructor to avoid an immediate re-render upon mount from `setState`. Someone shared this with me yesterday which I found quite informative about lifecycle methods and interactions with state:

https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9

But more importantly there might be some need to rethink how we're controlling or at least naming the initial content being passed in. Perhaps naming the prop as `defaultContent` (like `defaultValue` of base inputs) with a preference toward the state value in the `render`, or `initialContent` (`initialValue`) in defining the initial state.